python/lsst/pipe/tasks/processImage.py

```diff
@@ -132,101 +111,67 @@ def makeIdFactory(self, sensorRef):
     def getExposureId(self, sensorRef):
         raise NotImplementedError()
 
+    def getAstrometer(self):
+        """Return an astrometer for matching sources to the reference catalog"""
+        raise NotImplementedError("Not implemented in base class")
+
     @pipeBase.timeMethod
-    def process(self, dataRef, inputExposure, enableWriteSources=True):
+    def process(self, dataRef, inputExposure, idFactory=None, backgrounds=None, enableWriteSources=True):
         """Process an Image
-        
+
         @param dataRef: data reference that corresponds to the input image
-        @param inputExposure:  exposure to process
+        @param inputExposure: exposure to process
+        @param idFactory: afw.table.IdFactory to use for source catalog
+        @param backgrounds: afwMath.BackgroundList to be appended to
         @param enableWriteSources: if True then writing sources is allowed.
             Set False if you need to postprocess sources before writing them.
 
         @return pipe_base Struct containing these fields:
-        - postIsrExposure: exposure after ISR performed if calib.doIsr or config.doCalibrate, else None
-        - exposure: calibrated exposure (calexp): as computed if config.doCalibrate,
-            else as upersisted and updated if config.doDetection, else None
-        - calib: object returned by calibration process if config.doCalibrate, else None
+        - exposure: input exposure (as modified in the course of processing)
```

Comment:
Not sure exactly what that means -- "_will be_ modified in the course of processing"?